packages/verbatim/init.lua

```diff
@@ -13,20 +13,22 @@ function package:registerCommands ()
    self:registerCommand("verbatim", function (_, content)
       SILE.typesetter:pushVglue(6)
       SILE.typesetter:leaveHmode()
+      local lskip = SILE.settings:get("document.lskip") or SILE.types.node.glue()
```

Comment:
Why are we stashing the setting here, but creating a new glue from just the length instead of just reusing the glue we saved? For that matter why are we fiddling with lskip anyway? And why does this work without the super stretchy rskip.

So many questions, one has to wonder if I'm up too late and my brain just isn't processing answers that might be in front of my face.

Reply:
If I remember right, we'll want to kill stretch/shrink on lskip/rskip (line 23-24) but keep the fixed part, so verbatim in a block quote for instance now still obeys the fixed margins on both sides. Before it would be back to full line = we couldn't quote code properly in indented blocks. They reverted to full line width.

Reply:
As for: "And why does this work without the super stretchy rskip."

'Cause we handle break as paragraphs now = we have our back saved by the parfillskip.

Reply:
Roger and roger. Of course that makes sense.

Reply:
Now of course, one could argue that this whole `typesetter.parseppattern` stuff is _not_ sane. I'd have preferred the SIL parsers to know what a soft break is and insert that in a true AST, so the typesetter wouldn't have had to make assumptions on the way paragraphing is supposed to occur. But that's another "idiosyncrasy" of SIL that I'll leave to the future...

Reply:
How is that an idiosyncracy of the SIL parser? Don't we have exactly the same issue in XML?

Reply:
> Don't we have exactly the same issue in XML?

Yes we do, and this work indeed for SIL XML. 
But I (briefly) tackled it in #2111 (§ Thoughts on paragraphing).
Not all XML schemas would expect two blank lines to become paragraphs.
E.g. HTML doesn't.
Delegating the decision on what is a paragraph in the input schema/language to the typesetter seems off, it should perhaps be an inputter decision, in the scope of arbitrary XML support.
There are workarounds for implementers, but it's more a question of semantics and design.

Reply:
> Not all XML schemas would expect two blank lines to become paragraphs.

Of course not. How line breaks are handled is up to the content handler not XML itself. The schema might specify haw the content of a given tag should be handled. But for example given a SILE native schema I would expect my MWE above to treat newlines in `&lt;verbatim /&gt;` tags as breaks.

Hence why I'm suggesting this has nothing to do with the SIL language or parser. We don't really have an option of changing the parser to make newlines into nodes in the AST as long as we're trying to keep it interchangeable with an XML input.

> E.g. HTML doesn't. Delegating the decision on what is a paragraph in the input schema/language to the typesetter seems off

Even when the schema itself *says* that's how the content should be handled? We didn't invent XML or HTML.

Obviously we *could* just eschew SIL being sugar for an XML-ish schema and make it a totally unique language, but then I would question the sanity of having SIL as a markup language at all. We'd still have to have the typesetter handle this for actual XML, and SIL already isn't a *lightweight* markup language in the way Markdown/Djot/Asciidoc etc. are. So it would just be another complex idiosyncratic markup like USFM or others that only serve their niche and are frustrating to work with even in their niche.

> it should perhaps be an inputter decision, in the scope of arbitrary XML support.

I don't understand what you mean by this. What am I missing here?

Reply:
> I don't understand what you mean by this. What am I missing here?

Nevermind. It's orthogonal to verbatim, and _if time allows it,_ I'll eventually demonstrate a different approach to arbitrary XML input, based on some the ideas I discussed in #2111